src/Swarm/TUI/Model.hs

```diff
@@ -947,6 +950,15 @@ initAppState AppOpts {..} = do
 startGame :: (MonadIO m, MonadState AppState m) => Scenario -> ScenarioInfo -> Maybe FilePath -> m ()
 startGame = startGameWithSeed Nothing
 
+
+-- | Re-initialize the game from the stored reference to the current scenario
+restartGame :: (MonadIO m, MonadState AppState m) => m ()
+restartGame = do
+  g <- use uiState
+  let Just (scene, si) = _scenarioRef g
```

Comment:
Might be simpler to just say `s <- use $ uiState . scenarioRef` and then pattern-match on `s`.  We should also handle the case when the result is `Nothing` --- I know in theory that shouldn't happen but we still don't want the game to crash if it does happen for some reason.

Reply:
I've now handled the `Nothing` case by pushing the check all the way upstream to the population of buttons in the dialog.  The "Start over" button shall carry a reference to the scenario.   If the `scenarioRef` is `Nothing`, then the "Start over" button is not added.